### PR TITLE
fix authorization page, added new feature - floating label

### PR DESCRIPTION
**Pull request by iraskiba**
i was working on changePassword-page when i have to fix auth-page, therefore this pages is not ready.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -54,6 +54,7 @@
   },
   "devDependencies": {
     "file-loader": "^6.2.0",
+    "prettier": "^3.2.5",
```
Comment (AmidaWolf) [inline package.json:57]: Been better if we add eslint airbnb + prettier in the same time. But if this difficult  - u can do this next time.

File: src/components/Authorization/Authorization.css [hunk 2]
```diff
@@ -87,34 +88,32 @@ a{
 }
 
 .eye-icon {
+  padding:10px;
   position: absolute;
-  top: calc(36.6% - 8px);
-  left: calc(75% - 8px)
+  top: calc(41% - 8px);
+  left: calc(88% - 8px)
```
Comment (AmidaWolf) [inline src/components/Authorization/Authorization.css:94]: What about little explain, what happens here? Why we have these magic numbers like 41/88%? 
Another question for this item: why this item placed outside at input? We should place it inside the input/label and use position relative for parent and position absolute for this element. Also, if we have something like icon - we should create this like icon - with svg.

File: src/components/Authorization/Authorization.css [hunk 3]
```diff
@@ -225,66 +201,43 @@ a{
   }
 }
 
-@media (max-width: 660px) {
-  .eye-icon {
-    position: absolute;
-    top: 32.1%;
-    left: 73%;
+@media (max-width: 600px) {
```
Comment (AmidaWolf) [inline src/components/Authorization/Authorization.css:204]: We should use all media queries about screen size from one constant (I mean design system file with all constants like colors, containers, typography etc.).

File: src/components/Authorization/Authorization.jsx [hunk 4]
```diff
@@ -32,114 +33,168 @@ export const Authorization = ({primary}) => {
     event.preventDefault();
     setEmailError("");
     setPasswordError("");
-
+    let isValid = true;
     if (!email) {
       setEmailError("Это поле обязательно для заполнения");
```
Comment (AmidaWolf) [inline src/components/Authorization/Authorization.jsx:38]: DRY - don't repeat yourself. You should place this text to constant and reuse it. Better if we try to create object with properties like this:
```
const validationErrors = {
  required: "Это поле обязательно для заполнения",
  emailFormat: "Вы заполняете поле в неверном формате",
  passwordLength: "Пароль должен содержать минимум 8 символов",
}
```

Check this in another places.

File: src/components/Authorization/Authorization.jsx [hunk 5]
```diff
@@ -32,114 +33,168 @@ export const Authorization = ({primary}) => {
...
     }
-    // @ts-ignore
-    dispatch(authentication({
-      userEmail: email,
-      userPassword: password
-    }))
-  }
-    const onEmailChange = (event) => {
+    try {
+      // @ts-ignore
```
Comment (AmidaWolf) [inline src/components/Authorization/Authorization.jsx:49]: Any reason for  `// @ts-ignore`. Just try to learn how to fix it.

File: src/components/Authorization/Authorization.jsx [hunk 6]
```diff
@@ -32,114 +33,168 @@ export const Authorization = ({primary}) => {
...
+      await dispatch(
+        authentication({
+          userEmail: email,
+          userPassword: password,
+        }),
+      );
+    } catch (error) {
+      setEmailError("Неверный логин или пароль");
+      setPasswordError("Неверный логин или пароль");
```
Comment (AmidaWolf) [inline src/components/Authorization/Authorization.jsx:58]: Why we use something like this? For what?

File: src/components/Authorization/Authorization.jsx [hunk 7]
```diff
@@ -32,114 +33,168 @@ export const Authorization = ({primary}) => {
...
+
+              <div
+                className={`form-floating mb-3 input-field ${email ? "has-value" : ""}`}
+              >
+                <input
+                  className={`form-control ${emailError ? "is-invalid" : ""}`}
+                  id="floatingInput"
+                  type="email"
+                  name="email"
+                  placeholder=" "
```
Comment (AmidaWolf) [inline src/components/Authorization/Authorization.jsx:103]: If we don't need placeholder - just delete this property.

File: src/components/Authorization/Authorization.jsx [hunk 8]
```diff
@@ -32,114 +33,168 @@ export const Authorization = ({primary}) => {
...
-              <input className='input-field' type={passwordVisible ? 'text' : 'password'}
-                          id="pass"
-                          name="password"
-                          minLength={8}
-                          required
-                          value={password}
-                          placeholder="Пароль"
-                          onChange={onPasswordChange}
-                          style={{borderColor: passwordError ? "red" : "#282828", borderWidth: '1px'}}/>
+              <p className={"error " + (emailError ? "error-text" : "")}>
```
Comment (AmidaWolf) [inline src/components/Authorization/Authorization.jsx:91]: Don't use ternary operators like this. We just can do the same with this code:
```
emailError && "error-text"
```
But better if we just show this HTML element with ternary:
```
            {emailError  &&  <p className={"error " + "error-text"}>}
```

File: src/components/Authorization/Authorization.jsx [hunk 9]
```diff
@@ -32,114 +33,168 @@ export const Authorization = ({primary}) => {
...
       </div>
-      <Footer mainPageData={testMainPageData}/>
+      <Footer mainPageData={testMainPageData} />
     </div>
   );
 };
-
 Authorization.propTypes = {
   primary: PropTypes.bool,
 };
```
Comment (AmidaWolf) [inline src/components/Authorization/Authorization.jsx:194]: Why we use this? We should make this file .tsx and try to use typescript.

File: src/features/auth/ChangePassword/ChangePassword.tsx [hunk 10]
```diff
@@ -1,84 +1,145 @@
...
-                    </div>
-                </main>
-
-
+  return (
+    <>
+      <div className="container w-full md:h-full">
+        <Header />
+        <main>
+          <div className="md:w-72 mx-40 my-50 mt-10 sm:w-72">
```
Comment (AmidaWolf) [inline src/features/auth/ChangePassword/ChangePassword.tsx:86]: About styles - we should use only one variant for styling components. If we want to use css/scss - just replace these parts.

File: src/features/auth/ChangePassword/ChangePassword.tsx [hunk 11]
```diff
@@ -1,84 +1,145 @@
...
+            <Link to={Path.HOME}></Link>
+          </div>
+          {showNotification ? (
+            <NotificationComponent message={messages} />
+          ) : (
+            <div>
+              <Form onSubmit={handleSubmit(onSubmit)} className="m-5">
+                <h6
+                  style={{ marginBottom: "42px" }}
```
Comment (AmidaWolf) [inline src/features/auth/ChangePassword/ChangePassword.tsx:95]: Same. All styles should be placed to css/scss. Especially typography, for reusing in another places.

File: src/components/Authorization/Authorization.css [hunk 12]
```diff
@@ -14,9 +10,13 @@ input::placeholder {
   border-color: #282828;
```
Comment (Lansyfonseka) [inline src/components/Authorization/Authorization.css:10]: Better use .scss and variables $primary-dark-text

File: src/components/Authorization/Authorization.jsx (hunk 7 above)
Comment (iraskiba) [reply src/components/Authorization/Authorization.jsx:103]: we need a placeholder, when i delete this property  - inscription "Почта" disappear. I don't know why)

File: src/components/Authorization/Authorization.jsx (hunk 8 above)
Comment (iraskiba) [reply src/components/Authorization/Authorization.jsx:91]: fix

File: src/components/Authorization/Authorization.jsx (hunk 9 above)
Comment (iraskiba) [reply src/components/Authorization/Authorization.jsx:194]: in the next component i will try to use ts

File: src/features/auth/ChangePassword/ChangePassword.tsx (hunk 10 above)
Comment (iraskiba) [reply src/features/auth/ChangePassword/ChangePassword.tsx:86]: i will change this code

File: src/components/Authorization/Authorization.jsx [hunk 13]
```diff
@@ -1,145 +1,237 @@
...
+                  value={password}
+                  placeholder=" "
+                  onChange={onPasswordChange}
+                />
+                <label className="lab" htmlFor="pass">
+                  Пароль
+                </label>
+
+                {passwordVisible ? (
+                  <svg
```
Comment (Lansyfonseka) [inline src/components/Authorization/Authorization.jsx:134]: Use <svg> icon in main file bag idea. Icon should be isolated one file

File: src/components/Authorization/Authorization.jsx [hunk 14]
```diff
@@ -1,145 +1,237 @@
...
 
   const handleSubmit = async (event) => {
     event.preventDefault();
-    setEmailError("");
-    setPasswordError("");
-
+    setEmailError('');
+    setPasswordError('');
+    let isValid = true;
     if (!email) {
```
Comment (Lansyfonseka) [inline src/components/Authorization/Authorization.jsx:42]: One more variant realisation thils fetch. Use scss style. Like this:
const [passwordValid, setPasswordValid] = useState(true);
const [mailValid, setMailValid] = useState(true);
const handleSubmit = () => {
  ....
  setPasswordValid(!!password);
  setMailValid(!!mail);
  if (mailValid || passwordValid) return
 ...
}

<input className=`input ${passwordValid ? '' : 'invalid'}`

File: package.json (hunk 1 above)
Comment (iraskiba) [reply package.json:61]: fix

File: src/components/Authorization/Authorization.css (hunk 2 above)
Comment (iraskiba) [reply src/components/Authorization/Authorization.css]: fix

File: src/components/Authorization/Authorization.css (hunk 3 above)
Comment (iraskiba) [reply src/components/Authorization/Authorization.css:139]: fix

File: src/components/Authorization/Authorization.jsx (hunk 4 above)
Comment (iraskiba) [reply src/components/Authorization/Authorization.jsx]: fix

File: src/components/Authorization/Authorization.jsx (hunk 5 above)
Comment (iraskiba) [reply src/components/Authorization/Authorization.jsx]: fix

File: src/components/Authorization/Authorization.jsx (hunk 6 above)
Comment (iraskiba) [reply src/components/Authorization/Authorization.jsx]: fix

File: src/components/Authorization/Authorization.css (hunk 12 above)
Comment (iraskiba) [reply src/components/Authorization/Authorization.css]: fix

File: src/features/auth/ChangePassword/ChangePassword.tsx (hunk 11 above)
Comment (iraskiba) [reply src/features/auth/ChangePassword/ChangePassword.tsx:95]: i will change this code

File: src/components/Authorization/Authorization.jsx (hunk 13 above)
Comment (iraskiba) [reply src/components/Authorization/Authorization.jsx:134]: fix

**Review by Lansyfonseka — APPROVED** [review]
You should resolve merge conflict. Your changes correct

**Pull request merged**